experimental/segmenter/src/provider.rs

```diff
@@ -27,9 +28,34 @@ pub struct LineBreakDataV1<'data> {
     #[cfg_attr(feature = "provider_serde", serde(borrow))]
     pub property_table: LineBreakPropertyTable<'data>,
 
-    /// Rule table for line breaking.
+    /// Break state table for line breaking.
     #[cfg_attr(feature = "provider_serde", serde(borrow))]
-    pub rule_table: LineBreakRuleTable<'data>,
+    pub break_state_table: LineBreakStateTable<'data>,
+
+    /// Number of properties; should be the square root of the length of
+    /// [`Self::break_state_table`].
+    pub property_count: u8,
+
+    pub last_codepoint_property: i8,
+    pub sot_property: u8,
+    pub eot_property: u8,
+    pub complex_property: u8,
```

Comment:
Since line break rule uses EOT and complex (SA), we can re-use this. But we might have to re-think we add property defines in state table structure when we fix https://github.com/unicode-org/icu4x/issues/1637

Reply:
Thanks. Let's reconsider the way we add the properties definition in #1637.